src/test/java/com/teragrep/lsh_01/JsonSplittingTest.java

```diff
@@ -0,0 +1,201 @@
+        Assertions.assertEquals(1, payloads.size());
+        Assertions.assertTrue(payloads.get(0).contains(expected));
+    }
+
+    @Test
+    public void testJsonSplittingTwoMessages() {
+        String message1 = "{\"foo\": 1}";
+        String message2 = "{\"bar\": 2}";
+        String requestBody = "[\n" + message1 + ",\n" + message2 + "\n]";
+
+        NettyConfig nettyConfig = new NettyConfig();
+
+        HttpClient httpClient = HttpClient.newHttpClient();
+
+        HttpRequest request = HttpRequest
+                .newBuilder(URI.create("http://" + nettyConfig.listenAddress + ":" + nettyConfig.listenPort))
+                .POST(HttpRequest.BodyPublishers.ofString(requestBody))
+                .build();
+
+        CompletableFuture<HttpResponse<String>> response = httpClient
+                .sendAsync(request, HttpResponse.BodyHandlers.ofString());
+
+        int statusCode = Assertions.assertDoesNotThrow(() -> response.get().statusCode());
+        Assertions.assertEquals(200, statusCode);
+
+        List<String> payloads = this.relpServer.payloads();
+
+        String expected1 = message1.replaceAll("\\s", "");
+        String expected2 = message2.replaceAll("\\s", "");
+
+        // assert that payload was correctly split
+        Assertions.assertEquals(2, payloads.size());
+        Assertions.assertTrue(payloads.get(0).contains(expected1));
+        Assertions.assertFalse(payloads.get(0).contains(expected2));
+
+        Assertions.assertTrue(payloads.get(1).contains(expected2));
+        Assertions.assertFalse(payloads.get(1).contains(expected1));
+    }
+
+    @Test
+    public void testJsonSplittingThreeMessages() {
+        String message1 = "{\"foo\": 1}";
+        String message2 = "{\"bar\": 2}";
+        String message3 = "{\"foobar\": 3}";
+        String requestBody = "[\n" + message1 + ",\n" + message2 + "\n, \n" + message3 + "\n]";
+
+        NettyConfig nettyConfig = new NettyConfig();
+
+        HttpClient httpClient = HttpClient.newHttpClient();
+
+        HttpRequest request = HttpRequest
+                .newBuilder(URI.create("http://" + nettyConfig.listenAddress + ":" + nettyConfig.listenPort))
+                .POST(HttpRequest.BodyPublishers.ofString(requestBody))
+                .build();
+
+        CompletableFuture<HttpResponse<String>> response = httpClient
+                .sendAsync(request, HttpResponse.BodyHandlers.ofString());
+
+        int statusCode = Assertions.assertDoesNotThrow(() -> response.get().statusCode());
+        Assertions.assertEquals(200, statusCode);
+
+        List<String> payloads = this.relpServer.payloads();
+
+        String expected1 = message1.replaceAll("\\s", "");
+        String expected2 = message2.replaceAll("\\s", "");
+        String expected3 = message3.replaceAll("\\s", "");
+
+        // assert that payload was correctly split
+        Assertions.assertEquals(3, payloads.size());
+        Assertions.assertTrue(payloads.get(0).contains(expected1));
```

Comment:
please use String.equals() instead of String.contains() in all tests or even better Assertions.assertEquals(expected1, payloads.get(0));

Reply:
Switched to using rlo_06 for asserting the correctness of payloads with assertEquals. Just the message part, as the timestamp in the metadata is unique.